### PR TITLE
Redesigned top nav and footer

### PR DESCRIPTION
**Pull request by astephanus**
### Description
Ports the new design for site top banner header navigation, and site footer from the project website.
Also updates the menu item configuration data.

A note to code review. This is a port of a new design implementation from the project website where 1rem there != 1rem here. This repo uses a base theme that makes very greedy use of `!important` that needed to be forcefully overridden using `!important` despite its nature as an antipattern, and the base theme makes use of REM units in media queries which also needed to be accounted for. So, there are visibly bad practices at play in the changes in this PR.

This is an intermediate step towards impementing a new design entirely. In the interim we want to be at least aligned on having a header / footer that looks the same between the project website, and the documentation website, despite the kludges necessary to acheive this goal.

 [...]._

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
@ashwin-pc or @joshuarrrr - Can one of you please do a code review of this PR? Thanks.

**Comment by maezieLoo** [comment]
The BUILD FREELY tagline in the footer in mobile view overlays onto other text and there is a hyperlink attached to the tagline, which appears underlined in mobile view.

Request change: For mobile view, replace current logo/tagline with new vertical format svg (provided) and remove hyperlinks. SVG should use F4F8FB (OpenSky T3) as the graphic color.

LogoTagVertical


Screenshot 2023-10-04 at 11 13 29 AM

**Comment by astephanus** [comment]
@maezieLoo I have followed up on the footer fixes.

**Comment by joshuarrrr** [comment]
> @ashwin-pc or @joshuarrrr - Can one of you please do a code review of this PR? Thanks.

@hdhalter We don't have a lot of context on this. Which aspects of the PR would you like us to review? Just the custom styling, or something else?

**Comment by hdhalter** [comment]
Hi @joshuarrrr , @ashwin-pc - can you please review this PR from a technical standpoint? Aaron adjusted the style of the previous nav pane to match the style of the other website pages. The latest fix was to align the logo, Documentation in the gray pane, and the TOC: 
image. Thanks!

**Comment by BSFishy** [comment]
A few things I've noticed playing around with it locally:
 - There is a certain screen size where the header gets broken into multiple distinct lines:
   
   Screenshot (160)
 - When in mobile view, the collapsed sections' links are still clickable. I'll try to get a video, but you can validate by going into mobile view and hovering the mouse on the line between OpenSearchCon and Download while the OpenSearchCon section is collapsed
 - The Download and Documentation links don't work in mobile view, but I can see that they don't work on the main site, so it's probably just an issue copied over, but should still be fixed

**Comment by astephanus** [comment]
@BSFishy Nice find on that hidden menu item thing being clickable. I have resolved all mentioned issues.

File: _sass/custom/custom.scss [hunk 1]
```diff
@@ -1253,12 +1622,12 @@ main {
...
-    left: 0;
-    right: 0;
-    z-index: 2;
-    top: 0;
+    // padding: 1px 0;
+    // position: absolute;
+    // left: 0;
+    // right: 0;
+    // z-index: 2;
+    // top: 0;
```
Comment (BSFishy) [inline _sass/custom/custom.scss:1630]: In my experience, commenting things out is usually unnecessary. If we need to roll things back, we can just look in Git history.

**Review by BSFishy — COMMENTED** [review]
Nothing blocking, still need to run it with the latest changes for sanity testing

File: _sass/custom/custom.scss (hunk 1 above)
Comment (astephanus) [reply _sass/custom/custom.scss]: Oops! My appologies. The intention was to have deleted anything commented out during development prior to commit. This code and another commented out block have been removed.

**Review by BSFishy — APPROVED** [review]
LGTM

**Comment by hdhalter** [comment]
@astephanus - It looks like some menu options have changed since this was started. Please update. Thanks!

File: _data/top_nav.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ items:
     fragment: opensearchcon
     children:
       -
-        label: Register for OpenSearchCon!
+        label: Registration is closed
```
Comment (hdhalter) [inline _data/top_nav.yml:7]:
```suggestion
        label: 2024 - Stay informed
```

File: _data/top_nav.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ items:
...
         url: https://opensearchcon2023.splashthat.com
```
Comment (hdhalter) [inline _data/top_nav.yml:8]:
```suggestion
        url: https://opensearch.org/events/opensearchcon/
```

**Review by hdhalter — COMMENTED** [review]
Hi Aaron, I added a couple of comments, but the Community and Platform menus also need to be updated.

File: _data/top_nav.yml (hunk 3 above)
Comment (astephanus) [reply _data/top_nav.yml]: Updated menu items have been pushed.

**Review by hdhalter — APPROVED** [review]

**Pull request merged**